### DISCUSSION TITLE
Not able to Set Custom Roles and Permissions.

### DISCUSSION THREAD

**Discussion by sjugraj** [directus/directus#9286] [Archive]
I am using the latest rc.100 docker image of the App.
I have created a maintainer role with app access to some of the collections

![image](https://user-images.githubusercontent.com/70695640/139643306-9e4c9d06-3359-4fa4-b85b-29c6749d6e8b.png)

As you can see i have given the full permission to these collections.

In the same settings as show below i have added a user to it.
![image](https://user-images.githubusercontent.com/70695640/139643470-f2f676f8-fc84-41e7-b966-a90a116ad510.png)

But when i login via the user's credentials on separate browser.
the app shows no collections as shown below.
![image](https://user-images.githubusercontent.com/70695640/139643605-9a6478b1-bef2-41f5-bd11-4e32e560f694.png)

Even when the API for permissions returning the valid names as show below.
![image](https://user-images.githubusercontent.com/70695640/139643778-9a6a5650-2c8c-44a1-8200-aefeaa3e2a8b.png)

What i am doing wrong?

Just out of curiosity i have given all permissions to the role.
![image](https://user-images.githubusercontent.com/70695640/139649276-20d28bdb-dc29-4e7c-8eab-9b0798d8bb50.png)

Just not Admin Access
![image](https://user-images.githubusercontent.com/70695640/139649362-ace95edd-9a84-4f0b-9c35-f59d5935e31c.png)

Still not able to see the collections.
Something is weird happening in custom roles and permissions.

**Comment by azrikahar** [comment]
I just tested this earlier on rc.100 and it worked. Would you mind opening an issue for this? That way you'll be able to provide more details, and possibly the environment variables you're using etc which would be helpful to reproduce this on our end, or at the very least find out if it's a setup issue.

**Reply by sjugraj** [reply]
I think the issue happens when the collection is under a folder.

**Reply by sjugraj** [reply]
Can you try moving them inside a folder in the data model and then see. if a non admin user can view those in the navigation bar.

**Reply by azrikahar** [reply]
That explains it 👍 the folder is definitely a crucial detail here. There's an open issue regarding this problem: #9207. You can try the workaround offered in the issue at the moment while we look for a solution to this.

**Reply by sjugraj** [reply]
Thanks :+1: Should we close this discussion or move it to that issue?

**Reply by azrikahar** [reply]
No worries, we can leave this open for now until we solve that issue.
